### ISSUE TITLE
[prefer-inject] should not trigger for abstract classes

### ISSUE THREAD

**Issue by lichtenfeldn-cc** [angular-eslint/angular-eslint#2475]
**Description and reproduction of the issue**

Abstract class should not be included in the `prefer-inject` rule as they are only extended by other classes and therefore the parameters of the constructor are always provided by the extending class and not the dependency injection system.

```ts
@Component({ template: '' })
export abstract class BaseComponent {
  private _foo;

  constructor(foo: string) {  // triggers `prefer-inject`
    this._foo = foo;
  }
}

@Component({ template: '' })
export class MainComponent extends BaseComponent {
  constructor() {
    super('foo');
  }
}
```

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@angular-eslint/eslint-plugin` | `19.6.0` |
| `typescript-eslint`      | `5.7.2` |
| `ESLint`                        | `9.23.0` |
| `node`                          | `18.20.3` |

```sh
Angular CLI: 19.2.5
Node: 18.20.3
Package Manager: yarn 1.22.22
OS: linux x64

Angular: 19.2.4
... animations, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic
... upgrade

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1902.5
@angular-devkit/build-angular   19.2.5
@angular-devkit/core            19.2.5
@angular-devkit/schematics      19.2.5
@angular/cdk                    19.2.7
@angular/cli                    19.2.5
@angular/material               19.2.7
@schematics/angular             19.2.5
rxjs                            7.8.2
typescript                      5.7.3
webpack                         5.98.0
zone.js                         0.15.0
```

**Comment by JamesHenry** [comment]
I'm combining this with #2473

**Comment by JamesHenry** [comment]
Just because the code is hard to migrate automatically doesn’t make it suddenly make it a valid case for the rule, the option to the rule to allow abstract classes is still the solution is like to go with per the other issue
